File: python/no_766/no_766.py
from typing import List


class Solution:
    def isToeplitzMatrix(self, matrix: List[List[int]]) -> bool:
        m, n = len(matrix), len(matrix[0])
        for l in range(m):
            i, j = l, 0
            ini = matrix[i][j]
            while 0 <= i < m and 0 <= j < n:
                if ini != matrix[i][j]:
                    return False
                i += 1
                j += 1

        for l in range(n):
            i, j = 0, l
            ini = matrix[i][j]
            while 0 <= i < m and 0 <= j < n:
                if ini != matrix[i][j]:
                    return False
                i += 1
                j += 1

        return True

File: golang/problems/no_766/no_766.go
package no_766

func isToeplitzMatrix(matrix [][]int) bool {
	m, n := len(matrix), len(matrix[0])
	for l := 0; l < m ; l += 1 {
		i, j := l, 0
		ini := matrix[i][j]
		for i >= 0 && i < m && j >= 0 && j < n {
			if ini != matrix[i][j] {
				return false
			}

			i += 1
			j += 1
		}
	}

	for l := 0; l < n; l += 1 {
		i, j := 0, l
		ini := matrix[i][j]
		for i >= 0 && i < m && j >= 0 && j < n {
			if ini != matrix[i][j] {
				return false
			}

			i += 1
			j += 1
		}
	}
	return true
}